### DISCUSSION TITLE
Lazy tracking

### DISCUSSION THREAD

**Discussion by Trebol-a** [ultralytics/ultralytics#10275] [General]
Hi all, 
I am trying to track animals in a model trained with yolov8. I start the tracking like this: 

`        results = model.track(
            frame,
            save=False,
            conf=0.4,
            iou=0.4,
            verbose=False,
            agnostic_nms=True,
            persist=True,
            tracker="/home/trebol-a/entrenamiento/ultralytics/ultralytics/cfg/trackers/trailcamtrack.yaml"
        )
`

where my yaml say :
 
`tracker_type: bytetrack # tracker type, ['botsort', 'bytetrack']
track_high_thresh: 0.01 # threshold for the first association
track_low_thresh: 0.3 # threshold for the second association
new_track_thresh: 0.95 # threshold for init new track if the detection does not match any tracks
track_buffer: 250 # buffer to calculate the time when to remove tracks
match_thresh: 0.8 # threshold for matching tracks`

the detection is fine,  it starts at frame 0, but the tracking does not start until several seconds after the element is hidden for a few frames. Then the tracking starts and it does it with the ID : 3 (without passing through 0, 1, 2).

What is wrong?
It doesn't happen in all videos, in general the tracking works quite well. But in some videos it seems that it is difficult to start the id assignment.

Thank you

**Comment by pderrenger** [comment]
Hey there! 👋 It seems like you're encountering an interesting issue with the tracking delay and ID assignment in your YOLOv8 model. The configuration settings in your `trailcamtrack.yaml` seem fine at a glance.

One potential cause for the tracking to "lag" or delay could be related to the `track_buffer` and `new_track_thresh` settings in relation to the specific dynamics and characteristics of your videos. Since the behavior isn't consistent across all videos, video-specific traits (like the speed of the moving animals, occlusion, and frame rate) might influence the tracker's ability to initialize and assign IDs promptly.

A few suggestions to troubleshoot and possibly improve the tracking initiation:
- **Lower the `new_track_thresh`**: Lowering this value can make the tracker more eager to start new tracks, which might help in cases where the animals are not immediately tracked.
- **Adjust the `track_buffer`**: This controls how long the tracker waits before removing inactive tracks. Reducing it might help in resuming tracking quicker but be cautious as setting it too low might lead to losing track of animals too quickly during occlusions.

You could start by tweaking these parameters slightly to see if it improves the initial tracking latency.

If the issue persists, it could be valuable to visually inspect the frames leading up to the tracking start. Look for any anomalies or characteristics that might make tracking initiation difficult. This might provide further clues on the adjustments needed. 

Lastly, ensure that your model is accurately detecting the animals during those initial frames, as misdetections could also lead to the observed tracking behavior.

Would love to hear how these suggestions work out! Keep us posted. 🚀
